sql/moz-fx-data-shared-prod/contextual_services_derived/request_payload_suggest_v1/query.sql

```diff
@@ -0,0 +1,48 @@
+WITH urls AS (
+  SELECT
+    submission_timestamp,
+    SPLIT(error_message, "URL sent: ")[SAFE_OFFSET(1)] AS url,
+    udf_js.parse_sponsored_interaction(udf_js.extract_string_from_bytes(payload)) AS si
+  FROM
+    payload_bytes_error.contextual_services
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND error_type = 'SendRequest'
+),
+parsed AS (
+  SELECT
+    si.source,
+    REGEXP_EXTRACT(url, r'form-factor=(\w+)', 1) AS form_factor,
+    CASE
+      WHEN (url LIKE '%impressions%')
+        THEN 'impression'
+      ELSE 'click'
+    END AS send_event,
```

Comment:
`impressions` are only a part of the url for topsites impressions. using it here would flag suggest impressions as clicks. instead you can use `si.interactionType`